api/__tests__/index.test.js

```diff
@@ -0,0 +1,273 @@
+const { app, server } = require('../../server');
+const request = require('supertest');
+const promotionsJSON = require('../../data/promotions');
+
+afterAll(() => server.close());
+
+describe('Companies Endpoint', () => {
+  describe('/v1/companies/', () => {
+    it('Should respond to GET method', done => {
+      request(app)
+        .get('/v1/companies/')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(200, done);
+    });
+  });
+
+  describe('/companies/:companyId', () => {
+    it('Should respond to GET method', done => {
+      request(app)
+        .get('/v1/companies/1')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(res => {
+          expect(res.body).toEqual({ id: 1, name: 'Test Company' });
+        })
+        .expect(200, done);
+    });
+
+    it("Should respond with 404 when requested item doesn't exist", done => {
+      request(app)
+        .get('/v1/companies/9999')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(404, done);
+    });
+  });
+
+  describe('/companies/:companyId/promotions', () => {
+    const promotion = promotionsJSON.find(el => el.company_id === 1);
+
+    it('Should respond to GET method', done => {
+      return request(app)
+        .get('/v1/companies/1/promotions')
+        .set('Accept', 'application/json')
+        .expect(res => {
+          Object.keys(res.body[0]).forEach(key => {
```

Comment:
A better way of doing assertion here would be to pick one promotion object and do assertions on all the key:value pairs. I think it would be okay to just hardcode the strings, because this test is technically establishing a API Contract.

Look into `.toHaveProperty(keyPath, value)` assertion.